testing/e2e/suites/nginxIngressController.go

```diff
@@ -76,6 +76,30 @@ func nicTests(in infra.Provisioned) []test {
 					return fmt.Errorf("able to create NginxIngressController with invalid controllerNamePrefix '%s'", testNIC.Spec.ControllerNamePrefix)
 				}
 
+				testNIC = manifests.NewNginxIngressController("nginx-ingress-controller", "nginxingressclass")
+				testNIC.Spec.DefaultSSLCertificate.Secret.Name = "Invalid+@Name"
```

Comment:
https://github.com/Azure/aks-app-routing-operator/actions/runs/7757951112

e2e failing. Relevant line https://github.com/Azure/aks-app-routing-operator/actions/runs/7757951112/job/21158842340#step:8:1463


you aren't instantiating the DefaultSSLCertificate field anywhere here. You're trying to set a field on something that is nil

this should probably be testNIC.Spec.DefaultSSLCertificate := &DefaultSSLCertificate{
... <set fields here too>
}